brayns/engine/scene/ModelsOperations.h

```diff
@@ -20,42 +20,38 @@
 
 #pragma once
 
-#include <brayns/engine/geometry/GeometryView.h>
-#include <brayns/engine/model/Model.h>
-#include <brayns/engine/model/ModelComponents.h>
+#include "ModelManager.h"
 
 namespace brayns
 {
-/**
- * @brief Adds clipping geometry to the model (Affects the whole scene)
- */
-class ClippingComponent : public Component
+class ModelsOperations
 {
 public:
-    template<typename T>
-    ClippingComponent(T primitive)
-        : ClippingComponent(std::vector<T>{std::move(primitive)})
+    template<typename Component>
+    static void removeModelsWithComponent(ModelManager &models)
     {
-    }
+        static_assert(std::is_same_v<Component, std::decay_t<Component>>, "Only decayed types allowed");
 
-    template<typename T>
-    ClippingComponent(std::vector<T> primitives)
-        : _geometry(std::move(primitives))
-        , _view(_geometry)
-    {
-        _geometry.commit();
-        _view.commit();
-    }
+        auto &instances = models.getAllModelInstances();
 
-    virtual void onCreate() override
-    {
-        auto &model = getModel();
-        auto &group = model.getGroup();
-        group.setClipper(_view);
+        std::vector<uint32_t> ids;
+        ids.reserve(instances.size());
+
+        for (auto instance : instances)
+        {
+            auto &model = instance->getModel();
+            auto &components = model.getComponents();
+            if (components.has<Component>())
+            {
+                ids.push_back(instance->getID());
```

Comment:
Same question as in the cpp but it is not a big deal if it is simpler like that.